src/qt/guiutil.cpp

```diff
@@ -749,6 +749,20 @@ QString boostPathToQString(const fs::path &path)
     return QString::fromStdString(path.string());
 }
 
+QString ConnectionTypeToQString(const std::string& conn_type)
+{
+    // Outbound peer connections
+    if (conn_type == "outbound-full-relay") return "Outbound full relay";
+    if (conn_type == "block-relay-only") return "Outbound block relay";
+    if (conn_type == "manual") return "Outbound manual";
+    if (conn_type == "feeler") return "Outbound feeler";
+    if (conn_type == "addr-fetch") return "Outbound address fetch";
+    // Inbound peer connections
+    if (conn_type == "inbound") return "Inbound full relay";
+    if (conn_type == "inbound-block-relay") return "Inbound block relay";
+    return "";
```

Comment:
nit: Would be nice if this default fallback was replaced with an assert(false) and the above ifs with an enum-switch-case

Reply:
Agree, but ISTM that only works for integer types.

Reply:
`ConnectionType` is an enum class (implementation defined integer)

Reply:
Right, I wanted to do that but the `ConnectionType` enum wasn't exposed in CNodeStats. Ok, done.